### DISCUSSION TITLE
How to route all requests under a port to a handler

### DISCUSSION THREAD

**Discussion by monadbobo** [tokio-rs/axum#2175] [Q&A] [answered]
I'm writing an HTTP reverse proxy and I found out that code like this doesn't work (since it only routes /):

```
     let metrics_routes = Router::new()
             .route("/", get(http_get).post(http_post))
             .with_state(context. clone());
```

**Comment by davidpdrsn** [comment]
You can use https://docs.rs/axum/0.6.20/axum/handler/struct.HandlerService.html

**Reply by jplatte** [reply, marked as answer]
More specifically, see [this](https://docs.rs/axum/latest/axum/handler/struct.HandlerService.html#method.into_make_service) example if you want a single handler to serve all requests, and see [this](https://docs.rs/axum/latest/axum/routing/method_routing/struct.MethodRouter.html#method.into_make_service) example if you want to only route based on the request method, but not based on the path.

**Reply by jplatte** [reply]
Further, if you need one or two special-cased path routes and want the rest to go to the same handler, you can use [`Router::fallback`](https://docs.rs/axum/latest/axum/routing/struct.Router.html#method.fallback).
